config/url.go

```diff
@@ -0,0 +1,57 @@
+package config
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// AsURL returns the url.URL representation of the value associated with k or
+// panics if unable to do so.
+func AsURL(b Bucket, k string) *url.URL {
+	if v, ok := tryAsURL(b, k); ok {
+		return v
+	}
+
+	panic(fmt.Sprintf("%s is not defined", k))
+}
+
+// AsURLDefault returns the url.URL representation of the value associated with
+// k, or the default value v if k is undefined.
+func AsURLDefault(b Bucket, k, v string) *url.URL {
+	if v, ok := tryAsURL(b, k); ok {
+		return v
+	}
+
+	u, err := url.Parse(v)
```

Comment:
Why are we parsing the value as url for the second time?

Reply:
`v` is the default value in this case, it's only parsed if `k` is not set.

Reply:
Understood, thanks!